Source/Core/Core/HW/SI/SI_DeviceAMBaseboard.cpp

```diff
@@ -0,0 +1,2361 @@
+            dip_switch_0 &= ~0x80;
+          if (PadStatus.button & PAD_BUTTON_X)	// Service
+            dip_switch_0 &= ~0x40;
+          */
+
+          // Horizontal Scanning Frequency switch
+          // Required for F-Zero AX booting via Sega Boot
+          if (AMMediaboard::GetGameType() == FZeroAX ||
+              AMMediaboard::GetGameType() == FZeroAXMonster)
+          {
+            dip_switch_0 &= ~0x20;
+          }
+
+          // Disable camera in MKGP1/2
+          if (AMMediaboard::GetGameType() == MarioKartGP ||
+              AMMediaboard::GetGameType() == MarioKartGP2)
+          {
+            dip_switch_0 &= ~0x10;
+          }
+
+          data_out[data_offset++] = dip_switch_0;
+          data_out[data_offset++] = dip_switch_1;
+          break;
+        }
+        case GCAMCommand::SerialNumber:
+        {
+          NOTICE_LOG_FMT(SERIALINTERFACE_AMBB, "GC-AM: Command 0x11, {:02x} (READ SERIAL NR)",
+                         *data_in++);
+          data_out[data_offset++] = gcam_command;
+          data_out[data_offset++] = 16;
+          memcpy(data_out + data_offset, "AADE-01B98394904", 16);
+          data_offset += 16;
+          break;
+        }
+        case GCAMCommand::Unknown_12:
+          NOTICE_LOG_FMT(SERIALINTERFACE_AMBB, "GC-AM: Command 0x12, {:02x} {:02x}", *data_in++,
+                         *data_in++);
+          data_out[data_offset++] = gcam_command;
+          data_out[data_offset++] = 0x00;
+          break;
+        case GCAMCommand::Unknown_14:
+          NOTICE_LOG_FMT(SERIALINTERFACE_AMBB, "GC-AM: Command 0x14, {:02x} {:02x}", *data_in++,
+                         *data_in++);
+          data_out[data_offset++] = gcam_command;
+          data_out[data_offset++] = 0x00;
+          break;
+        case GCAMCommand::FirmVersion:
+          NOTICE_LOG_FMT(SERIALINTERFACE_AMBB, "GC-AM: Command 0x15, {:02x} (READ FIRM VERSION)",
+                         *data_in++);
+          data_out[data_offset++] = gcam_command;
+          data_out[data_offset++] = 0x02;
+          // 00.26
+          data_out[data_offset++] = 0x00;
+          data_out[data_offset++] = 0x26;
+          break;
+        case GCAMCommand::FPGAVersion:
+          NOTICE_LOG_FMT(SERIALINTERFACE_AMBB, "GC-AM: Command 0x16, {:02x} (READ FPGA VERSION)",
+                         *data_in++);
+          data_out[data_offset++] = gcam_command;
+          data_out[data_offset++] = 0x02;
+          // 07.06
+          data_out[data_offset++] = 0x07;
+          data_out[data_offset++] = 0x06;
+          break;
+        case GCAMCommand::RegionSettings:
+        {
+          // Used by SegaBoot for region checks (dev mode skips this check)
+          // In some games this also controls the displayed language
+          NOTICE_LOG_FMT(SERIALINTERFACE_AMBB,
+                         "GC-AM: Command 0x1F, {:02x} {:02x} {:02x} {:02x} {:02x} (REGION)",
+                         *data_in++, *data_in++, *data_in++, *data_in++, *data_in++);
```

Comment:
Function argument evaluation order is unspecified in C++. This produces unspecified behavior.
i.e. The logged values might end up in reverse order.

FYI, this is a warning with `-Wunsequenced`.

There are more instances of this problem.